src/EFCore.Relational/Query/RelationalQueryableMethodTranslatingExpressionVisitor.cs

```diff
@@ -1098,9 +1274,51 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCallExp
             && selectExpression.Having == null
             && selectExpression.Orderings.Count == 0
             && selectExpression.Tables.Count == 1
-            && selectExpression.Tables[0] is TableExpression)
+            && selectExpression.Tables[0] is TableExpression expression)
+        {
+            tableExpression = expression;
+
+            return true;
+        }
+
+        tableExpression = null;
+        return false;
+    }
+
+    // TODO: Update this documentation.
+    /// <summary>
+    ///     Validates if the current select expression can be used for execute update operation or it requires to be pushed into a subquery.
```

Comment:
(also update delete above)

```suggestion
    ///     Checks whether the current select expression can be used as-is for an execute update operation, or whether it must be pushed down into a subquery.
```

Reply:
You wrote this in first place. :trollface: 

I updated delete docs. For update, I still need to think how to do "pushdown"

Reply:
:grimacing: